From 199367ceeebd3677a742a4d5bc6d97b15137f524 Mon Sep 17 00:00:00 2001
From: bxu10x <bingx.xu@intel.com>
Date: Wed, 25 Jul 2018 15:30:50 +0800
Subject: [PATCH] Do not check the config_voice_capable when make a call

Setting "config_voice_capable" flag which is used in
telecom to display call screen. Setting the flag causes
CTS failures. Skip the Check for config_voice_capable in
automotive device when make a bt call.

Change-Id: I8c0271099e94e33708016b83ad9e981fd81d3a65
Tracked-On: OAM-75510
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Signed-off-by: bxu10x <bingx.xu@intel.com>
---
 .../android/server/telecom/TelecomServiceImpl.java    | 11 +++++++----
 .../telecom/components/UserCallIntentProcessor.java   |  7 +++++--
 2 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/src/com/android/server/telecom/TelecomServiceImpl.java b/src/com/android/server/telecom/TelecomServiceImpl.java
index ea55e63..51daa44 100644
--- a/src/com/android/server/telecom/TelecomServiceImpl.java
+++ b/src/com/android/server/telecom/TelecomServiceImpl.java
@@ -409,11 +409,14 @@ public class TelecomServiceImpl {
             try {
                 Log.startSession("TSI.rPA");
                 synchronized (mLock) {
-                    if (!mContext.getApplicationContext().getResources().getBoolean(
-                            com.android.internal.R.bool.config_voice_capable)) {
-                        Log.w(this,
+                    if (!mContext.getPackageManager()
+                        .hasSystemFeature(PackageManager.FEATURE_AUTOMOTIVE)) {
+                        if (!mContext.getApplicationContext().getResources().getBoolean(
+                                com.android.internal.R.bool.config_voice_capable)) {
+                            Log.w(this,
                                 "registerPhoneAccount not allowed on non-voice capable device.");
-                        return;
+                            return;
+                        }
                     }
                     try {
                         enforcePhoneAccountModificationForPackage(
diff --git a/src/com/android/server/telecom/components/UserCallIntentProcessor.java b/src/com/android/server/telecom/components/UserCallIntentProcessor.java
index 0c8525f..0e495da 100644
--- a/src/com/android/server/telecom/components/UserCallIntentProcessor.java
+++ b/src/com/android/server/telecom/components/UserCallIntentProcessor.java
@@ -19,6 +19,7 @@ package com.android.server.telecom.components;
 import android.app.admin.DevicePolicyManager;
 import android.content.Context;
 import android.content.Intent;
+import android.content.pm.PackageManager;
 import android.net.Uri;
 import android.os.UserHandle;
 import android.os.UserManager;
@@ -82,8 +83,10 @@ public class UserCallIntentProcessor {
     public void processIntent(Intent intent, String callingPackageName,
             boolean canCallNonEmergency, boolean isLocalInvocation) {
         // Ensure call intents are not processed on devices that are not capable of calling.
-        if (!isVoiceCapable()) {
-            return;
+        if (!mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_AUTOMOTIVE)) {
+            if (!isVoiceCapable()) {
+                return;
+            }
         }
 
         String action = intent.getAction();
-- 
2.17.1

